tools/packaging/kata-deploy/scripts/kata-deploy.sh

```diff
@@ -635,11 +627,32 @@ function main() {
 
 		case "$action" in
 		install)
+			if [[ "$runtime" =~ ^(k3s|k3s-agent|rke2-agent|rke2-server)$ ]]; then
+			       if [ ! -f "$containerd_conf_tmpl_file" ] && [ -f "$containerd_conf_file" ]; then
+				       cp "$containerd_conf_file" "$containerd_conf_tmpl_file"
+			       fi
+			       # Only set the containerd_conf_file to its new value after
+			       # copying the file to the template location
+			       containerd_conf_file="${containerd_conf_tmpl_file}"
+			elif [[ "$runtime" =~ ^(k0s-worker|k0s-controller)$ ]]; then
+			       touch "$containerd_conf_file"
+			elif [[ "$runtime" == "containerd" ]]; then
+			       if [ ! -f "$containerd_conf_file" ] && [ -d $(dirname "$containerd_conf_file") ] && [ -x $(command -v containerd) ]; then
+					containerd config default > "$containerd_conf_file"
+			       fi
+			fi
+
 			install_artifacts
 			configure_cri_runtime "$runtime"
 			kubectl label node "$NODE_NAME" --overwrite katacontainers.io/kata-runtime=true
 			;;
 		cleanup)
+			if [[ "$runtime" =~ ^(k3s|k3s-agent|rke2-agent|rke2-server)$ ]]; then
+			       if [ -f "$containerd_conf_file_backup" ]; then
```

Comment:
To be honest, I think we don't actually need.

Let me put this PR as a DRAFT, and I will double check it.
Thanks for the catch!

Reply:
It's right, because it just means that containerd will be using the "default" configuration.

And I have removed the check on the cleanup action side.

Reply:
So, once the CI passes I will go ahead and merge this one. :-)